azure-boards-common/src/main/java/com/synopsys/integration/azure/boards/common/http/AzureApiVersionAppender.java

```diff
@@ -0,0 +1,43 @@
+/**
+ * azure-boards-common
+ *
+ * Copyright (c) 2020 Synopsys, Inc.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.synopsys.integration.azure.boards.common.http;
+
+import static com.synopsys.integration.azure.boards.common.http.AzureHttpService.AZURE_API_VERSION_QUERY_PARAM_NAME;
+
+import org.apache.commons.lang3.StringUtils;
+
+public class AzureApiVersionAppender {
+    public String appendApiVersion(String spec, String apiVersion) {
+        char queryParamSeparator = '?';
+        if (StringUtils.contains(spec, queryParamSeparator)) {
+            queryParamSeparator = '&';
+        }
+        spec += queryParamSeparator;
```

Comment:
Instead of modifying the provided String shouldn't we create a new one in this method?

Reply:
Given that string are immutable, that IS what occurs anyway. But I can make it more explict.